File: frontend/src/features/message/http/response/MessageConversationSummary.ts
import type { UserIdentity } from '@/features/user/model/UserIdentity'
import type { Username } from '@/features/user/model/Username'
import type { MessageConversationId } from '@/features/message/model/MessageConversationId'

export type MessageConversationSummary = {
  id: MessageConversationId
  otherUser: UserIdentity
  lastMessagePreview: string | null
  lastMessageSenderUsername: Username | null
  lastMessageAt: string
  unreadCount: number
}

File: backend/src/main/scala/domains/message/model/response/MessageConversationSummary.scala
package domains.message.model.response

import domains.message.model.*

import domains.user.model.Username
import domains.user.model.UserIdentity

import java.time.Instant

final case class MessageConversationSummary(
  id: MessageConversationId,
  otherUser: UserIdentity,
  lastMessagePreview: Option[String],
  lastMessageSenderUsername: Option[Username],
  lastMessageAt: Instant,
  unreadCount: Int
)
